src/components/CompositionModel.tsx

```diff
@@ -0,0 +1,106 @@
+import "grommet/grommet-hpinc.min.css";
+import * as React from "react";
+import { connect } from "react-redux";
+import { getApiList, selectApi, formSubmit } from "actions/configForm";
+import { setCompositionModel } from "actions/compositionModel";
+import { Parser } from "../utils/parser";
+import { DrawComponent } from "../utils/drawComponent";
+import { initialization } from "actions/components";
+import SideBar from "components/SideBar";
+import * as Split from "grommet/components/Split";
+import * as Box from "grommet/components/Box";
+import Footer from "components/Footer";
+
+const mapStateToProps = (state) => {
+    return {
+        currentComponent: state.components.currentComponent,
+        getDrawComponent: () => {
+            let initialized = state.components.initialized;
+            if (!initialized) {
+                return null;
+            } else {
+                let currentComponent = state.components.currentComponent;
+                let componentProperties = state.components.componentProperties;
+                return componentProperties[currentComponent].drawComponent;
+            }
+        }
+    };
+};
+
+const mapDispatchToProps = (dispatch, ownProps) => {
+    return {
+        initialization: (componentProperties, currentComponent, projectName) => {
+            dispatch(initialization(componentProperties, currentComponent, projectName));
+        }
+    };
+};
+
+class CompositionModel extends React.Component<any, any> {
+    constructor(props: any) {
+        super(props);
+        this.getContainersForGraphs = this.getContainersForGraphs.bind(this);
+    }
+
+    componentDidMount() {
+        const props = this.props;
+        let comps = props.compositionModel.components;
+        let componentProperties = {};
+        for (let i = 0; i < comps.length; i++) {
+            let parser = new Parser();
```

Comment:
better get a comp[i] as parameter in the constructor.
then just.

parse.parse();